server/athenian/api/controllers/search_controller/search_prs.py

```diff
@@ -238,6 +248,12 @@ async def _search_pr_digests(
     return pr_digests
 
 
+def _apply_stages_filter(pr_facts: pd.DataFrame, stages: Collection[str]) -> pd.DataFrame:
+    masks = pr_facts_stages_masks(pr_facts)
+    filter_mask = pr_stages_mask(stages)
+    return pr_facts.iloc[np.nonzero(masks & filter_mask)]
```

Comment:
Never use `iloc` in production code.
```suggestion
    return pr_facts.take(np.flatnonzero(masks & filter_mask))
```